modules/gridmap/grid_map.cpp

```diff
@@ -136,6 +136,14 @@ void GridMap::_get_property_list(List<PropertyInfo> *p_list) const {
 	p_list->push_back(PropertyInfo(Variant::DICTIONARY, "data", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_STORAGE));
 }
 
+void GridMap::_validate_property(PropertyInfo &p_property) const {
+	if (p_property.name == "cell_layout" && cell_shape == CELL_SHAPE_SQUARE) {
+		p_property.usage ^= PROPERTY_USAGE_READ_ONLY;
```

Comment:
```suggestion
		p_property.usage |= PROPERTY_USAGE_READ_ONLY;
```
Assuming we want to force set it here, not toggle it